src/examples/signal_generator.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+#include <hyro/common/Time.h>
+#include <hyro/core/StateMachine.h>
+#include <hyro/widgets/WidgetCollection.h>
+#include <hyro/factory/CommandFactory.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+#include <../include/hyro/SignalGeneratorComponent.h>
+#include <../include/hyro/DigitalConverterComponent.h>
+
+using namespace hyro;
+using namespace signal_generator;
+using namespace digital_converter;
+using namespace std::string_literals;
+
+int main(void)
+{
+
+  //State machine creation
+  StateMachine signal_generator_sm(std::make_shared<SignalGeneratorComponent>("/generator"_uri));
+  StateMachine digital_converter_sm(std::make_shared<DigitalConverterComponent>("/converter"_uri));
+
+  std::string generator_configuration;
+  std::string converter_configuration;
+
+  // Configuration Strings
+  generator_configuration = "{"
+                            "outputs: {"
+                            "analog_output: { protocol: 'grpc', ip: '0.0.0.0',   port: '50051' },"
+                            "fix_dynamic: { protocol: 'api'},"
+                            "}"
+                            "}";
+
+  converter_configuration = "{"
+                            "inputs: {"
+                            "analog_input: { protocol: 'auto' }"
+                            "},"
+                            "outputs: {"
+                            "digital_output: { protocol: 'api' }"
+                            "}"
+                            "}";
+
+  //input_proto = "grpc";
+  
+  // Dynamic Property
+  signal_generator_sm.init(ComponentConfiguration(generator_configuration));
+  digital_converter_sm.init(ComponentConfiguration(converter_configuration));
+
+  auto remote_endpoint = ConnectionConfiguration("{ analog_input: { endpoint: '/generator/analog_output' } }");
+
+  signal_generator_sm.start();
+  digital_converter_sm.start();
+
+  signal_generator_sm.connect();
+  digital_converter_sm.connect(remote_endpoint);
+
+  signal_generator_sm.check();
+  digital_converter_sm.check();
+
+  float amp,freq;
+  bool cosine;
+
+  DynamicPropertyAccess dynamic_property_access("/generator"_uri);
+
+  dynamic_property_access.get<float>("amplitude", amp);
+  dynamic_property_access.get<float>("frequency", freq);
+  dynamic_property_access.get<bool>("cosine", cosine);
+
+  std::cout << "Amplitude: "<< amp << " Frequency: " << freq << " Waveform: " << (cosine?"Cosine":"Sine") <<std::endl;
```

Comment:
One suggestion! The code from line 69 to 84 could be in a loop inside a thread. In this way you can change the dynamic parameters dynamically. :ocean: